source/Calamari/Integration/Scripting/WindowsPowerShell/PowerShellBootstrapper.cs

```diff
@@ -58,20 +60,30 @@ public static string FormatCommandArguments(string bootstrapFile, CalamariVariab
             {
                 commandArguments.Append($"-Version {customPowerShellVersion} ");
             }
-            var executeWithoutProfile = variables[SpecialVariables.Action.PowerShell.ExecuteWithoutProfile];
-            bool noProfile;
-            if (bool.TryParse(executeWithoutProfile, out noProfile) && noProfile)
+            var debug = variables[SpecialVariables.Action.PowerShell.ExecuteWithoutProfile];
+            bool shouldDebug;
+            if (bool.TryParse(debug, out shouldDebug) && shouldDebug)
             {
                 commandArguments.Append("-NoProfile ");
             }
             commandArguments.Append("-NoLogo ");
             commandArguments.Append("-NonInteractive ");
             commandArguments.Append("-ExecutionPolicy Unrestricted ");
-            var escapedBootstrapFile = bootstrapFile.Replace("'", "''");
-            commandArguments.AppendFormat("-Command \"Try {{. {{. '{0}' -OctopusKey '{1}'; if ((test-path variable:global:lastexitcode)) {{ exit $LastExitCode }}}};}} catch {{ throw }}\"", escapedBootstrapFile, encryptionKey);
+
+            var filetoExecute = IsDebuggingEnabled(variables)
+                ? debuggingBootstrapFile.Replace("'", "''")
+                : bootstrapFile.Replace("'", "''");
+
+            commandArguments.AppendFormat("-Command \"Try {{. {{. '{0}' -OctopusKey '{1}'; if ((test-path variable:global:lastexitcode)) {{ exit $LastExitCode }}}};}} catch {{ throw }}\"", filetoExecute, encryptionKey);
             return commandArguments.ToString();
         }
 
+        private static bool IsDebuggingEnabled(CalamariVariableDictionary variables)
+        {
+            var powershellDebugMode = variables[SpecialVariables.Action.PowerShell.DebugMode];
```

Comment:
I think you should test if the value is "truthy" not just if it has a value - people will expect to be able to set it to "false"